sources/assets/Xenko.Core.Assets/Templates/TemplateManager.cs

```diff
@@ -59,7 +59,12 @@ public static void Unregister(ITemplateGenerator generator)
         /// <returns>A sequence containing all registered template descriptions.</returns>
         public static IEnumerable<TemplateDescription> FindTemplates(PackageSession session = null)
         {
-            var packages = session?.Packages.Concat(ExtraPackages) ?? ExtraPackages;
+            IEnumerable<Package> packages = session?.Packages ?? ExtraPackages;
+            if (packages != ExtraPackages)
```

Comment:
I am afraid the extra `if` branch (depending on which part of ternary operator is taken) is not so easy to understand at a quick glance.

What about using `Distinct()` to remove duplicates?
`var packages = session?.Packages.Concat(ExtraPackages).Distinct() ?? ExtraPackages;`

Reply:
Done, however it needs a comparer since the packages are not object equals (in fact the `ExtraPackages` packages have empty `Meta` data.
I avoided Distinct the first time round as I'm semi-conscious of the hashset it builds behind the scenes, however since I think this isn't called frequently it should be ok.